torch/utils/data/dataloader.py

```diff
@@ -90,6 +93,12 @@ def __iter__(self):
             yield None
 
 
+def _sharding_worker_init_fn(worker_init_fn, worker_id):
+    if worker_init_fn is not None:
+        worker_init_fn(worker_id)
+    torch.utils.data.backward_compatibility.private_worker_init_fn(
+        worker_id)
```

Comment:
We should call `private_worker_init_fn` before custom `worker_init_fn`, just in case users want to override the default behavior.